### ISSUE TITLE
win32a problem building with MinGW (undefined reference)

### ISSUE THREAD

**Issue by GitMensch** [Bill-Gray/PDCurses#12]
Doing `cd win32a && make -f gccwin32.mak DLL=Y` results in:

```
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/addch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/addchstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/addstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/attr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/beep.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/bkgd.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/border.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/clear.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/color.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/delch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/deleteln.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/deprec.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/getch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/getstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/getyx.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/inch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/inchstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/initscr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/inopts.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/insch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/insstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/instr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/kernel.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/keyname.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/mouse.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/move.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/outopts.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/overlay.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/pad.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/panel.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/printw.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/refresh.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/scanw.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/scr_dump.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/scroll.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/slk.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/termattr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/terminfo.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/touch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/util.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/window.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/debug.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32a/pdcclip.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32a/pdcdisp.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32a/pdcgetsc.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32a/pdckbd.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32a/pdcscrn.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32a/pdcsetsc.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32a/pdcutil.c
echo LIBRARY pdcurses > pdcurses.def
echo EXPORTS >> pdcurses.def
cat ../exp-base.def >> pdcurses.def
gcc pdcurses.def -Wl,--out-implib,pdcurses.a -shared -o pdcurses.dll addch.o addchstr.o addstr.o attr.o beep.o bkgd.o border.o clear.o color.o delch.o deleteln.o deprec.o getch.o getstr.o getyx.o inch.o inchstr.o initscr.o inopts.o insch.o insstr.o instr.o kernel.o keyname.o mouse.o move.o outopts.o overlay.o pad.o panel.o printw.o refresh.o scanw.o scr_dump.o scroll.o slk.o termattr.o terminfo.o touch.o util.o window.o debug.o pdcclip.o pdcdisp.o pdcgetsc.o pdckbd.o pdcscrn.o pdcsetsc.o pdcutil.o pdcurses.def
pdcdisp.o:pdcdisp.c:(.text+0x318): undefined reference to `CreateFontIndirectA@4'
pdcdisp.o:pdcdisp.c:(.text+0x38d): undefined reference to `ChooseFontA@4'
pdcdisp.o:pdcdisp.c:(.text+0x3ae): undefined reference to `CommDlgExtendedError@0'
pdcdisp.o:pdcdisp.c:(.text+0x8ed): undefined reference to `SetTextColor@8'
pdcdisp.o:pdcdisp.c:(.text+0x905): undefined reference to `SetBkColor@8'
pdcdisp.o:pdcdisp.c:(.text+0x963): undefined reference to `SelectObject@8'
pdcdisp.o:pdcdisp.c:(.text+0xa17): undefined reference to `ExtTextOutW@32'
pdcdisp.o:pdcdisp.c:(.text+0xa89): undefined reference to `CreatePen@12'
pdcdisp.o:pdcdisp.c:(.text+0xaa1): undefined reference to `SelectObject@8'
pdcdisp.o:pdcdisp.c:(.text+0xb2b): undefined reference to `MoveToEx@16'
pdcdisp.o:pdcdisp.c:(.text+0xb58): undefined reference to `LineTo@12'
pdcdisp.o:pdcdisp.c:(.text+0xbb4): undefined reference to `MoveToEx@16'
pdcdisp.o:pdcdisp.c:(.text+0xbda): undefined reference to `LineTo@12'
pdcdisp.o:pdcdisp.c:(.text+0xc00): undefined reference to `SelectObject@8'
pdcdisp.o:pdcdisp.c:(.text+0xc11): undefined reference to `DeleteObject@4'
pdcdisp.o:pdcdisp.c:(.text+0xd64): undefined reference to `SelectObject@8'
pdcdisp.o:pdcdisp.c:(.text+0xdf8): undefined reference to `MoveToEx@16'
pdcdisp.o:pdcdisp.c:(.text+0xe0e): undefined reference to `LineTo@12'
pdcdisp.o:pdcdisp.c:(.text+0xe43): undefined reference to `MoveToEx@16'
pdcdisp.o:pdcdisp.c:(.text+0xe59): undefined reference to `LineTo@12'
pdcdisp.o:pdcdisp.c:(.text+0xe93): undefined reference to `MoveToEx@16'
pdcdisp.o:pdcdisp.c:(.text+0xea9): undefined reference to `LineTo@12'
pdcdisp.o:pdcdisp.c:(.text+0xf61): undefined reference to `DeleteObject@4'
d:/programme/entwicklung/mingw/bin/../lib/gcc/mingw32/4.8.1/../../../../mingw32/bin/ld.exe: pdcdisp.o: bad reloc address 0x20 in section `.eh_frame'
collect2.exe: error: ld returned 1 exit status
make: *** [pdcurses.dll] Error 1
```

Any hints how to solve this issue?

**Comment by GitMensch** [comment]
I've rechecked `win32`, works like a charm:

~~~
$ make -f gccwin32.mak DLL=Y
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/addch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/addchstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/addstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/attr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/beep.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/bkgd.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/border.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/clear.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/color.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/delch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/deleteln.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/deprec.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/getch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/getstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/getyx.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/inch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/inchstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/initscr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/inopts.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/insch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/insstr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/instr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/kernel.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/keyname.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/mouse.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/move.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/outopts.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/overlay.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/pad.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/panel.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/printw.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/refresh.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/scanw.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/scr_dump.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/scroll.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/slk.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/termattr.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/terminfo.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/touch.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/util.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/window.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../pdcurses/debug.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32/pdcclip.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32/pdcdisp.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32/pdcgetsc.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32/pdckbd.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32/pdcscrn.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32/pdcsetsc.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD ../win32/pdcutil.c
echo LIBRARY pdcurses > pdcurses.def
echo EXPORTS >> pdcurses.def
cat ../exp-base.def >> pdcurses.def
gcc pdcurses.def -Wl,--out-implib,pdcurses.a -shared -o pdcurses.dll addch.o addchstr.o addstr.o attr.o beep.o bkgd.o border.o clear.o color.o delch.o deleteln.o deprec.o getch.o getstr.o getyx.o inch.o inchstr.o initscr.o inopts.o insch.o insstr.o instr.o kernel.o keyname.o mouse.o move.o outopts.o overlay.o pad.o panel.o printw.o refresh.o scanw.o scr_dump.o scroll.o slk.o termattr.o terminfo.o touch.o util.o window.o debug.o pdcclip.o pdcdisp.o pdcgetsc.o pdckbd.o pdcscrn.o pdcsetsc.o pdcutil.o pdcurses.def
cp pdcurses.a panel.a
gcc -O2 -Wall -I.. -DPDC_DLL_BUILD -otestcurs.exe ../demos/testcurs.c pdcurses.dll
gcc -O2 -Wall -I.. -DPDC_DLL_BUILD -onewdemo.exe ../demos/newdemo.c pdcurses.dll
gcc -O2 -Wall -I.. -DPDC_DLL_BUILD -oxmas.exe ../demos/xmas.c pdcurses.dll
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD -I../demos -otuidemo.o ../demos/tuidemo.c
gcc -c -O2 -Wall -I.. -DPDC_DLL_BUILD -I../demos -otui.o ../demos/tui.c
gcc  -otuidemo.exe tuidemo.o tui.o pdcurses.dll
gcc -O2 -Wall -I.. -DPDC_DLL_BUILD -ofirework.exe ../demos/firework.c pdcurses.dll
gcc -O2 -Wall -I.. -DPDC_DLL_BUILD -optest.exe ../demos/ptest.c pdcurses.dll
gcc -O2 -Wall -I.. -DPDC_DLL_BUILD -orain.exe ../demos/rain.c pdcurses.dll
gcc -O2 -Wall -I.. -DPDC_DLL_BUILD -oworm.exe ../demos/worm.c pdcurses.dll
strip *.exe
~~~

Seems win32a misses a header or library.

Next check with `make -f mingwin32.mak DLL=Y`: results in

    make: *** No rule to make target `..\exp-base.def', needed by `pdcurses.def'.  Stop.

`make -f mingwin32.mak DLL=Y clean` results in:

~~~
del *.o
/bin/sh: del: command not found
make: [clean] Error 127 (ignored)
del *.exe
/bin/sh: del: command not found
make: [clean] Error 127 (ignored)
del *.dll
/bin/sh: del: command not found
make: [clean] Error 127 (ignored)
del *.def
/bin/sh: del: command not found
make: [clean] Error 127 (ignored)
del pdcurses.dll *.a pdcurses.def
/bin/sh: del: command not found
make: [clean] Error 127 (ignored)
~~~

`make -f mingwin32.mak` results in:

~~~
copy pdcurses.a panel.a
make: copy: Command not found
make: *** [pdcurses.a] Error 127
~~~

**Comment by GitMensch** [comment]
mingwin32.mak in win32 does not work:

    make -f mingwin32.mak DLL=Y
    make: *** No rule to make target `..\exp-base.def', needed by `pdcurses.def'.  Stop.

mingwin32.mak in win32 does not work when run from an msys shell as it uses copy and del, both not available as msys uses `sh`, not `cmd`.

**Comment by GitMensch** [comment]
Rechecked with MSYS2 package (both MINGW32 and MINGW64). mingw32.mak won't work as there is no `x86_64-w64-mingw32-gcc` or `x86_64-w64-mingw32-gcc.exe` -it would be `gcc`, therefore going on with gccwin32.mak. Running this with ` make -f gccwin32.mak DLL=Y` leads (with MINGW32 from MSYS2) to: 

~~~
pdcdisp.o:pdcdisp.c:(.text+0xc14): undefined reference to `__imp_SelectObject'
pdcdisp.o:pdcdisp.c:(.text+0xc1f): undefined reference to `__imp_DeleteObject'
pdcdisp.o:pdcdisp.c:(.text+0xd16): undefined reference to `__imp_SelectObject'
pdcdisp.o:pdcdisp.c:(.text+0xddb): undefined reference to `__imp_MoveToEx'
pdcdisp.o:pdcdisp.c:(.text+0xde2): undefined reference to `__imp_LineTo'
pdcdisp.o:pdcdisp.c:(.text+0xe10): undefined reference to `__imp_MoveToEx'
pdcdisp.o:pdcdisp.c:(.text+0xe1e): undefined reference to `__imp_LineTo'
pdcdisp.o:pdcdisp.c:(.text+0xe46): undefined reference to `__imp_MoveToEx'
pdcdisp.o:pdcdisp.c:(.text+0xe54): undefined reference to `__imp_LineTo'
pdcdisp.o:pdcdisp.c:(.text+0xe84): undefined reference to `__imp_MoveToEx'
pdcdisp.o:pdcdisp.c:(.text+0xe92): undefined reference to `__imp_LineTo'
pdcdisp.o:pdcdisp.c:(.text+0xf43): undefined reference to `__imp_DeleteObject'
pdcscrn.o:pdcscrn.c:(.text+0xe6): undefined reference to `__imp_GetStockObject'
pdcscrn.o:pdcscrn.c:(.text+0xf5): undefined reference to `__imp_SelectObject'
pdcscrn.o:pdcscrn.c:(.text+0x168): undefined reference to `__imp_Rectangle'
pdcscrn.o:pdcscrn.c:(.text+0x189): undefined reference to `__imp_Rectangle'
pdcscrn.o:pdcscrn.c:(.text+0x269): undefined reference to `__imp_SelectObject'
pdcscrn.o:pdcscrn.c:(.text+0x27f): undefined reference to `__imp_GetTextMetricsA'
pdcscrn.o:pdcscrn.c:(.text+0x29c): undefined reference to `__imp_DeleteObject'
pdcscrn.o:pdcscrn.c:(.text+0x974): undefined reference to `__imp_GetStockObject'
~~~

Any ideas?

Note in general: the makefiles should use `rm -f` instead of `rm` as this will neither ask nor error if the file doesn't exist.

**Comment by Bill-Gray** [comment]
Um.  In `mingwin32.mak`, this line:

`ifneq (,$(findstring CYGWIN,$(uname_S)))`

i's supposed to figure out that you're running on Cygwin and drop the prefixes to `gcc`.  If you're getting prefixes on `gcc`,  that test is failing.

As a test,  try changing the `ifneq` to `ifeq`.  Flipping that check ought to cause everything else to follow correctly: no prefix to `gcc`,  use `rm` and `cp` instead of `del` and `copy`,  etc.  If that works... we may have to make that line read

`ifeq ($(CYGWIN),Y)

and require that you run,  for example,  `make -f mingwin32.mak DLL=Y CYGWIN=Y` etc.

`gccwin32.mak` is obsolete;  I'm surprised it got as far as it did before emitting errors.

**Comment by GitMensch** [comment]
If the gccwin32.mak is obsolete: please delete it, it is confusing.

... and with the following change mingwin32.mak actually works with MINGW32 and MINGW64 (may drop the `in32`?), too:

```diff
 ifneq (,$(findstring CYGWIN,$(uname_S)))
 	#  Insert Cygwin-specific changes here
 	ON_WINDOWS = 1
 endif
+ifneq (,$(findstring MINGW32_NT,$(uname_S)))
+	#  Insert MINGW32-specific changes here
+	ON_WINDOWS = 1
+endif
+ifneq (,$(findstring MINGW64_NT,$(uname_S)))
+	#  Insert MINGW64-specific changes here
+	ON_WINDOWS = 1
+endif
```

**Comment by Bill-Gray** [comment]
I should have said that `gccwin32.mak` is headed for obsolescence,  once I got confirmation that `mingwin32.mak` was working.  Which,  it appears,  we're close to having happen.

Am I correct in thinking we can remove the four lines for "Cygwin-specific changes",  and that they aren't doing anything under Cygwin anyway?

**Comment by GitMensch** [comment]
> Am I correct in thinking we can remove the four lines for "Cygwin-specific changes", and that they aren't doing anything under Cygwin anyway?

No, they are needed for cygwin (I've tested with MINGW32 and MINGW64 before). I've just retested cygwin and cygwin64 and both work (if the four lines stay in). As this new Makefile is used for these four systems it should be renamed in any case. And I think gccwin32.mak is actually a nice name  - therefore I suggest to simply update this one (if it is older) and remove the "new" one.

**Comment by GitMensch** [comment]
win32a builds fine now with MINGW64, with and without DLL=Y.
But using `WIDE=Y` leads to

~~~
gcc -c -O4 -Wall -pedantic -I.. -DPDC_WIDE -DPDC_DLL_BUILD ../win32a/pdcdisp.c
../win32a/pdcdisp.c: In function 'PDC_get_logical_font':
../win32a/pdcdisp.c:203:9: warning: implicit declaration of function '_tcscpy'  -Wimplicit-function-declaration]
         _tcscpy( PDC_font_name, _T("Courier New"));
         ^~~~~~~
gcc -c -O4 -Wall -pedantic -I.. -DPDC_WIDE -DPDC_DLL_BUILD ../win32a/pdcscrn.c
../win32a/pdcscrn.c: In function 'get_app_name':
../win32a/pdcscrn.c:1154:9: warning: implicit declaration of function '_wcslwr' [-Wimplicit-function-declaration]
         _wcslwr( buff + 1);
         ^~~~~~~
../win32a/pdcscrn.c:1098:24: warning: implicit declaration of function 'wcslwr' [-Wimplicit-function-declaration]
    #define my_tcslwr   wcslwr
                        ^
../win32a/pdcscrn.c:1098:24: note: in definition of macro 'my_tcslwr'
    #define my_tcslwr   wcslwr
                        ^~~~~~
pdcdisp.o:pdcdisp.c:(.text+0x390): undefined reference to `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x390): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x494): undefined reference to `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x494): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x4ab): undefined reference to `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x4ab): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x51a): undefined reference to `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x51a): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x6ab): undefined reference to `_tcscpy'
pdcdisp.o:pdcdisp.c:(.text+0x6ab): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_tcscpy'
pdcscrn.o:pdcscrn.c:(.text+0x30d): undefined reference to `wcslwr'
pdcscrn.o:pdcscrn.c:(.text+0x30d): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `wcslwr'
pdcscrn.o:pdcscrn.c:(.text+0x44f): undefined reference to `_wcslwr'
pdcscrn.o:pdcscrn.c:(.text+0x44f): relocation truncated to fit: R_X86_64_PC32 against undefined symbol `_wcslwr'
~~~
The above is identical when using cygwin64, therefore it looks like a general win32 issue.


Strangely, using `UTF8=Y` doesn't lead to `-DPDC_FORCE_UTF8` in the MSYS2 environment but works in the cygwin64 and cygwin environment... I guess this is a bug of this environment.


In general, `DEBUG=Y` shouldn't strip anything, see pull request #39

**Comment by GitMensch** [comment]
If the `WIDE` issue is solved and someone tested the cross-compiling we can delete gccwin32.mak in both win32 and win32a.

**Comment by GitMensch** [comment]
`WIDE` issue is resolved with 5dee41f. Only cross-compilation is missing.
